### PR TITLE
Create new issue backend

### LINKED ISSUE #21: Create/View new issue
Issues can be used to keep track of bugs, enhancements, or other requests.

Mandatory attributes of an issue: **title** (250 chars)

Optional attributes of an issue: **description** (1500 chars), **labels** (fixed set of values: bug, enhancement, backend, frontend, test)

tasks
- [ ] create backend classes
- [ ] write the code to save a new issue to the database
- [ ] create rest/graphql api for creating an issue

### PR DESCRIPTION
**Pull request by codergo7**

### CODE REVIEW

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 1]
```diff
@@ -1,4 +1,43 @@
...
+@Table(name = "issue")
 public class IssueEntity {
+
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    private String tittle;
```
Comment (Avk44) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java]: suggestion
--> should be written like this "title"

File: backend/src/main/java/com/kodstar/backend/model/dto/Issue.java [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+import lombok.Getter;
+
+import java.util.Set;
+
+@Getter
+@AllArgsConstructor
 public class Issue {
+
+    private Long id;
+    private String tittle;
```
Comment (Avk44) [inline backend/src/main/java/com/kodstar/backend/model/dto/Issue.java]: suggestion
--> should be written like this "title"

Comment (codergo7) [reply backend/src/main/java/com/kodstar/backend/model/dto/Issue.java]: will be fixed the typo. Thanks

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java (hunk 1 above)
Comment (codergo7) [reply backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java]: will be fixed the typo. Thanks

File: backend/src/main/java/com/kodstar/backend/model/enums/Label.java [hunk 3]
```diff
@@ -1,4 +1,6 @@
 package com.kodstar.backend.model.enums;
 
 public enum Label {
+
+    TODO, STORY
```
Comment (kopfadam) [inline backend/src/main/java/com/kodstar/backend/model/enums/Label.java]: 1. modify enums with values like TODO("todo")
2. insert labels' names as described --> labels (fixed set of values: bug, enhancement, backend, frontend, test)

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 4]
```diff
@@ -1,4 +1,43 @@
...
+
+    @NotBlank
+    private String description;
+
+    @NotBlank
+    @Enumerated(EnumType.STRING)
+    private Set<Label> labels;
```
Comment (kopfadam) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java:35]: description and labels attributes should be optional

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 5]
```diff
@@ -1,4 +1,43 @@
...
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    private String tittle;
+
+    @NotBlank
+    private String description;
```
Comment (Avk44) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java:31]:
```Suggestion
         Add @Length(....) or @Column(length = ..) validations to attributes
```

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java (hunk 1 above)
Comment (Avk44) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java:28]:
```Suggestion
     Add @Length(....) or @Column(length = ..) validations to attributes
  ```

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 6]
```diff
@@ -1,4 +1,43 @@
...
+
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    private String tittle;
+
+    @NotBlank
```
Comment (kopfadam) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java]: description should be optional

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 7]
```diff
@@ -1,4 +1,43 @@
...
+    private String description;
+
+    @NotBlank
```
Comment (kopfadam) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java]: labels should be optional

File: backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java [hunk 8]
```diff
@@ -1,4 +1,43 @@
...
 public class IssueEntity {
+
+    @Id
+    @NotNull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
+    private String tittle;
+
```
Comment (Avk44) [inline backend/src/main/java/com/kodstar/backend/model/entity/IssueEntity.java:24]:
```Suggestion
 Add @Length(..) or @Column(length = ..) validations to attributes
```

**Review by kopfadam — CHANGES_REQUESTED** [review]
request changes test

**Review by Avk44 — APPROVED** [review]

**Review by kopfadam — APPROVED** [review]

**Pull request merged**